Gemfile

```diff
@@ -32,7 +32,8 @@ gem 'jsbundling-rails' # Bundle and transpile JavaScript
 gem 'i18n-js', '3.9.0' # A library to provide the I18n translations on the Javascript
 # gem 'devise-i18n' # Translations for Devise
 
-gem 'psych', '< 4.0'
+# Fix Ruby 3.1 that uses psych version 4.0 and conflicts with rails-template
```

Comment:
this comment does not reflect the actual situation. What does it mean by "conflicts with rails-template"?

Reply:
I tried to create a rails app without a template using Ruby `3.2.1` without downgrading Psych to version 3, and everything worked fine. So, I think there will be some issue that does not work properly in the template.

I might be misunderstanding the root issue in this case.

Reply:
You mean you created the project without using the Rails template, and it worked, right?
It's because Rails template uses the Figaro gem, and it will break if you use Ruby 3.1 and above (which requires psych v4).
So, even if you initialized the project without the Rails template, later, if you add the Figaro gem (or any other gem that depends on `psynch` but has not been updated to work with psynch v4), it will break again.
Check this issue for more details
https://github.com/nimblehq/rails-templates/issues/316.

So instead of forcing `psych < 4.0`, I would suggest, either
- Removing the Figaro gem, replacing by https://github.com/bkeepers/dotenv. The end purpose is to allow the loading of different configuration per environment (we're making these changes in Rails template https://github.com/nimblehq/rails-templates/pull/400)
- Downgrading Ruby to a lower version ( Ruby 3.0.x)

Reply:
also to clarify: in the end, did you use the Rails-template to initialize this project?